tests/server/socksd.c

```diff
@@ -169,7 +169,7 @@ static unsigned char byteval(char *value)
 static unsigned short shortval(char *value)
 {
   unsigned long num = strtoul(value, NULL, 10);
-  return (short)num & 0xffff;
+  return (unsigned short)num & 0xffff;
```

Comment:
If no compiler complains on this **and** it actually ANDs the value with `0xffff ` isn't this rather your linter that is overly picky?

Reply:
Well, the function is converting a string representation of a number into a `long`, and then is potentially truncating it (yes yes I know what the `&` does). So even though compilers may not complain about this, it is still implicit, and it's better to be explicit.

Reply:
>  it is still implicit, and it's better to be explicit.

Only to a certain extent.

It is also good to not write more code than we need to and to make the code easier on the eye for the reader. And typecasts do make a rather big dent in the flow when you read code I think. In my mind, code should have as few typecasts as possible.

Also: the function just above `shortval` in the same file is called `byteval` and it has the exact same logic but for an `unsigned char`. What made your linter be happy with that?